src/IECoreScene/Primitive.cpp

```diff
@@ -78,10 +81,27 @@ Imath::Box3f Primitive::bound() const
 		if( p )
 		{
 			const vector<V3f> &pp = p->readable();
-			for( size_t i=0; i<pp.size(); i++ )
-			{
-				result.extendBy( pp[i] );
-			}
+			using RangeType = tbb::blocked_range< const V3f* >;
+			tbb::this_task_arena::isolate( [ & result, & pp ]{
+				result = tbb::parallel_reduce( RangeType( pp.data(), pp.data() + pp.size(), 1000 ), Imath::Box3f(),
+				[]( const RangeType& range, const Imath::Box3f& value ) -> Imath::Box3f
+				{
+					Imath::Box3f b( value );
+					for( const V3f
+						*      it    = range.begin(),
+						*const itEnd = range.end(); it != itEnd; ++it )
```

Comment:
`for( const V3f &point : range )`

Now that range-for exists, I automatically assume that a for loop with pointer arithmetic in will contain some custom logic rather than just be visiting every element in the range.

Reply:
done